pkg/generators/models/testdata/cases/oneof_mapped_discriminator/generated/model_error.go

```diff
@@ -39,6 +39,31 @@ type Error struct {
 }
 
 var EmptyErrorError = fmt.Errorf("empty data is not an Error")
+var NotErrorError = fmt.Errorf("could not convert to type Error")
+
+type ErrorValidationNotNil struct{}
```

Comment:
It might be helpful if the generated name ended with `Rule`, i struggled to read it at first when i was reading the model_container code.

Do we need the notNil error rule? wouldn't that just be the default Validate function? I guess the naming is clearer this way maybe.